gdnative-core/src/core_types/color.rs

```diff
@@ -48,6 +65,157 @@ impl Color {
         }
     }
 
+    #[inline]
+    pub fn blend(&self, other: &Color) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_blend)(self.sys(), other.sys()) })
+    }
+
+    #[inline]
+    pub fn contrasted(&self) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_contrasted)(self.sys()) })
+    }
+
+    #[inline]
+    pub fn darkened(&self, amount: f32) -> Color {
+        Color::from_sys(unsafe { (get_api().godot_color_darkened)(self.sys(), amount) })
+    }
+
+    #[inline]
+    pub fn gray(&self) -> f32 {
+        // Implemented as described in godot docs
+        (self.r + self.b + self.g) / 3.0
+    }
+
+    #[inline]
+    pub fn inverted(&self) -> Color {
+        // Implementation as described in godot docs.
+        Color {
+            r: 1.0f32 - self.r,
+            g: 1.0f32 - self.g,
+            b: 1.0f32 - self.b,
+            a: self.a,
+        }
+    }
```

Comment:
These two methods have a simple implementation, but is there a reason to not call the GDNative API? Or where do we draw the border?

For the case with `to_abgr64`, where the wrong type is returned by GDNative, it clearly makes sense.

Reply:
I think it's fine to inline simple implementations on small types to avoid the FFI overhead. We do this for `Vector2` and `Vector3`, and `Color` is essentially a `Vector4`. I don't see how this is problematic.

Reply:
Ok, I think that's a good boundary 👍🏼 
And yes, let's leave the other methods as is for now, no need to change what's not broken.

The only small risk is that if a Godot method ever got a small change in semantics or bugfix, it wouldn't be carried over, but for most of the methods this is unlikely. 

One subtlety I could imagine is how the `[0.0, 1.0]` float range is mapped to the integer interval `[0, 255]` -- just multiplying by 255 does not create even bins. All values in `[0.0, 1/255]` are mapped to 0, but only the exact value `1.0` is mapped to 255. In other words, in a uniform distribution, it is infinitely improbable for a color to have a component with value `255` (in theory at least).

Reply:
@Bromeon Uniform [0, 1] distributions do exist. The distribution used by Godot's `RandomNumberGenerator` class is one: https://github.com/godotengine/godot/blob/28f56e2cbf03a164741f2eade17f9515f887482c/core/math/random_pcg.h#L90

Reply:
Absolutely. My point was more that _if_ Godot ever decides to change how this is implemented, we will be deviating slightly and could cause subtle bugs for users.

But I don't think it's something worthwhile to worry about, we could even add tests for such cases (later, not in this PR).

Reply:
That's a good point. I implemented these by hand initially because they were trivial and I wasn't sure how to call the API. If there's no significant performance hit, it'd probably be better to call the api. I'll test it later when I'm back at my PC.

Reply:
No, I didn't mean you should change it. As toasteater said, it's OK in these cases 🙂 

I think you've had enough obstacles in this PR already -- if we _really_ feel this could become a problem one day, we can simply add regression tests where we compare the built-in vs. the API function.

Reply:
Thanks for all of the information/discussion. I'll work on implementing a basic regression test for the custom implementation for this type and include it in another PR a little later.